### ISSUE TITLE
Configure PSSA from Start-EditorServices

### ISSUE THREAD

**Issue by csc027** [PowerShell/PowerShellEditorServices#2001]
### Summary
How do I configure the underlying PSScriptAnalyzer (PSSA) instance to use custom rules from the starting point `Start-EditorServices` for use in non-Visual Studio Code editors?

### Description
I have a working PSES setup working for my Neovim setup, providing completion and linting for the default ruleset.  However, I currently do not know how to configure custom PSSA rules for the PSSA instance bundled with PSES.

I know for VS Code, you could change the settings in your workspace, as documented [here](https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/using-scriptanalyzer?view=ps-modules#using-custom-rules-in-visual-studio-code).

To be clear, I was able to get those rules working via implicit settings documented [here](https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/using-scriptanalyzer?view=ps-modules#implicit).  However, I have many different PowerShell repositories in my use case.  It would be onerous to save a settings file each and every repository.

So, is it possible to configure the underlying PSSA instance's settings to explicitly use custom rules from either `Start-EditorServices` or from making specific changes to the environment that PSSA runs from?

**Comment by andschwa** [comment]
I think you would have to send those same settings that VS Code uses over LSP using whatever method your LSP client in Neovim provides to do so. In my experience this can be kind of tricky depending on the client, but here's the [docs](https://neovim.io/doc/user/lsp.html) I found for Neovim's LSP client.

**Comment by andschwa** [comment]
Ah ha, specifically you'll want to look at [`vim.lsp.start_client()`](https://neovim.io/doc/user/lsp.html#vim.lsp.start()), and pass it a map under the `settings` key in the start config.

**Comment by andschwa** [comment]
Here is a completely random example of that I found on GitHub for you: https://github.com/ChristianChiarulli/nvim/blob/1631262e8df1de2ad0ecfd5f7dffd9c4476d7933/ftplugin/java.lua#L119

**Comment by csc027** [comment]
Thank you so much!  I was tearing my hair out trying to figure out how to pass the values to the server.  For some reason, I was so stuck thinking about how to configure it via the entry point, that it didn't event occur to me that I should pass the settings through the LSP protocol.

**Comment by csc027** [comment]
Also, how useful would it be to add this somewhere to the documentation somewhere, perhaps somwhere in `docs/guide`?

**Comment by andschwa** [comment]
It would be a great addition. I have work planned for this month to make using PSES from Emacs/Vim etc. (as in other LSP clients that are not VS Code) _much_ easier, and will include some tested documentation on configuration. Because yes, LSP can be so confusing. It's fantastic tech compared to where we were at a decade ago, but I remember the pain the first time I tried to get Emacs with an LSP client to send a setting to OmniSharp (a C# LSP server)...it's the only reason I remembered enough to know what to look for here!
